steward/src/cellars.rs

```diff
@@ -23,12 +23,48 @@ pub async fn get_gas_price() -> Result<U256, Error> {
     CellarGas::etherscan_standard().await.map_err(|e| e.into())
 }
 
-fn parse_cellar_id(cellar_id: &str) -> CellarId {
-    let parts = cellar_id.split_at(cellar_id.chars().position(|c| c == ':').unwrap());
-    let address: H160 = H160::from_slice(parts.1.as_bytes());
+fn parse_cellar_id(cellar_id: &str) -> Result<CellarId, String> {
```

Comment:
Is there a specific error type we would prefer to be returning here rather than a String?

Reply:
It's probably fine in this case, but in general returning a basic type like String obfuscates the error condition and makes it more clunky to write code around handling it. Generally, a function should be acting like a black box -- if there is a more appropriate error type it can return, then a caller who needs a string representation of the error can format that at their discretion.